SwiftReflector/SwiftInterfaceReflector/SwiftInterfaceReflector.cs

```diff
@@ -1103,6 +1116,259 @@ List<XElement> GatherInheritance (Type_inheritance_clauseContext context, bool f
+
+		Version GetVersionAfter (List<string> pieces, int indexAfter)
+		{
+			var colonIndex = ColonAfter (pieces, indexAfter);
+			if (colonIndex < 0)
+				return new Version (0, 0);
+			var start = colonIndex + 1;
+			var end = start + 1;
+			while (end < pieces.Count && pieces [end] != ",")
+				end++;
+			var versionPieces = pieces.GetRange (start, end - start);
+			return VersionPiecesToVersion (versionPieces);
+		}
+
+		int ColonAfter (List<string> pieces, int start)
+		{
+			for (int i = start + 1; i < pieces.Count; i++) {
+				if (pieces [i] == ":")
+					return i;
+				if (pieces [i] == ",")
+					return -1;
+			}
+			return -1;
+		}
+
+		bool TargetVersionIsGreaterOrEqual (List<string> versionPieces)
+		{
+			var expectedVersion = VersionPiecesToVersion (versionPieces);
+			return TargetVersionIsGreaterOrEqual (expectedVersion);
+		}
+
+		bool TargetVersionIsGreaterOrEqual (Version expectedVersion)
+		{
+			var compiledVersionStr = PlatformVersionFromModuleFlags ();
+			if (String.IsNullOrEmpty (compiledVersionStr))
+				return true; // no version, I guess it's good?
+			var compiledVersion = new Version (compiledVersionStr);
+			return expectedVersion >= compiledVersion;
+		}
+
+		bool TargetVersionIsLessOrEqual (List<string> versionPieces)
+		{
+			var expectedVersion = VersionPiecesToVersion (versionPieces);
+			return TargetVersionIsLessOrEqual (expectedVersion);
+		}
+
+		bool TargetVersionIsLessOrEqual (Version expectedVersion)
+		{
+			var compiledVersionStr = PlatformVersionFromModuleFlags ();
+			if (String.IsNullOrEmpty (compiledVersionStr))
+				return true; // no version, I guess it's good?
+			var compiledVersion = new Version (compiledVersionStr);
+			return expectedVersion <= compiledVersion;
+		}
+
+		Version VersionPiecesToVersion (List<string> pieces)
+		{
+			var sb = new StringBuilder ();
+			for (int i = 0; i < pieces.Count; i++) {
+				if (pieces [i] == "." && i + 1 < pieces.Count && pieces [i + 1] == "*")
+					break;
+				sb.Append (pieces [i]);
+			}
+			return new Version (sb.ToString ());
+		}
+
+		IEnumerable<XElement> AvailableAttributes (XElement attributes)
+		{
+			if (attributes == null)
+				return Enumerable.Empty<XElement> ();
+			return attributes.Descendants (kAttribute).Where (el => el.Attribute (kName).Value == kAvailable);
+
+		}
+
+		List<string> AttrbuteParameters (XElement attribute)
+		{
+			return attribute.Descendants (kAttributeParameter).Select (at =>
+				at.Attribute (kValue)?.Value ?? "").ToList ();
+		}
+
+		bool PlatformMatches (string platform)
+		{
+			var currentPlatform = PlatformFromModuleFlags ();
+			switch (platform) {
+			case "*":
+			case "swift":
+				return true;
+			case "iOS":
+			case "iOSApplicationExtension":
+				return currentPlatform.StartsWith ("ios", StringComparison.Ordinal);
+			case "macOS":
+			case "macOSApplicationExtension":
+			case "macCatalyst":
+			case "macCatalystExtension":
+				return currentPlatform.StartsWith ("macos", StringComparison.Ordinal);
```

Comment:
macOS and MacCatalyst are different platforms, the latter will use version numbers that match iOS since it's largely based on that code

Reply:
OK - I'll open up an issue on that. I wasn't able to generate that tag from the compiler easily so I guessed (incorrectly).

Reply:
And I addressed it - not so bad.